### PR TITLE
Introduce More SZF Entities

### PR DESCRIPTION
**Pull request by Batfoxkid**
Basically I have a running fork and a request I had from map makers was to know when events took place in SZF such as changing map music. Would be great to have this as an official addition.

### szf_laststand
`Trigger` to activate last stand music
`OnUser1` when there is only one survivor left alive (activator is the client)
`OnUser2` when the last control point is being captured

### szf_zombierage / szf_panic_event
`OnUser1` when a frenzy begins
`OnUser2` when a frenzy ends

### szf_zombietank / szf_tank
`OnUser1` when a tank spawns (activator is the client)
`OnUser2` when a tank dies (activator is the client)

### szf_weapon_common
A weapon that is only common, such as spawn weapons but aren't infinite

### szf_weapon_uncommon
A weapon that is only uncommon, such a rare weapon when the rare cap is hit

### CODE REVIEW

**Comment by Batfoxkid** [comment]
For example a map uses it's own music when a frenzy/tank starts and needs to return to it's normal music when a frenzy/tank ends.

**Comment by FortyTwoFortyTwo** [comment]
> For example a map uses it's own music when a frenzy/tank starts and needs to return to it's normal music when a frenzy/tank ends.


Fair enough, though now i'm unsure which is best route to go for triggering relays when frenzy/tank end, either one of those:
- Start calls Trigger and FireUser1, end calls FireUser2
- Start calls Trigger, end calls FireUser1
- Start calls Trigger, end calls Trigger to new relay szf_zombierage_end/szf_zombietank_end etc.

I feel first suggestion is best to go for, @haxtonsale thoughts?

**Comment by Batfoxkid** [comment]
Alright, added a stock `FireRelay`, added rare cap info_target, and support for having multiple Tanks at once via the map or commands.

File: addons/sourcemod/scripting/szf/pickupweapons.sp [hunk 1]
```diff
@@ -88,13 +90,21 @@ public Action Event_WeaponsRoundStart(Event event, const char[] name, bool dontB
...
 				SetRandomWeapon(iEntity, WeaponRarity_Rare);
 			}
+			case WeaponType_Common:
+			{
+				SetRandomWeapon(iEntity, eWeaponsRarity_Common);
+			}
+			case WeaponType_Uncommon:
+			{
+				SetRandomWeapon(iEntity, eWeaponsRarity_Uncommon);
+			}
```
Comment (FortyTwoFortyTwo) [inline addons/sourcemod/scripting/szf/pickupweapons.sp:100]: cool cool github not doing a build test. `eWeaponsRarity_` should be `WeaponsRarity_`.

File: addons/sourcemod/scripting/superzombiefortress.sp [hunk 2]
```diff
@@ -439,6 +439,7 @@ public void OnPluginStart()
 	g_bNoDirectorTanks = false;
 	g_bNoDirectorRages = false;
 	g_bDirectorSpawnTeleport = false;
+	g_iRareCap = 15;
```
Comment (FortyTwoFortyTwo) [inline addons/sourcemod/scripting/superzombiefortress.sp:442]: Use `MAX_RARES` define as that still exists

File: addons/sourcemod/scripting/szf/stocks.sp [hunk 3]
```diff
@@ -320,6 +320,18 @@ stock bool IsMapSZF()
 	return false;
 }
 
+stock void FireRelay(const char[] sInput, const char[] sTargetName1, const char[] sTargetName2 = "")
```
Comment (FortyTwoFortyTwo) [inline addons/sourcemod/scripting/szf/stocks.sp:323]: Missed readding activator for last stand and tank?

**Pull request merged**